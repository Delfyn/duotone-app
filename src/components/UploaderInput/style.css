.panel-file-picker {
  cursor: pointer;
}

.panel-file-picker h2 {
  font-weight: 600;
}

.panel-file-picker span {
  font-weight: 300;
}

.panel-file-picker.has-file h2:hover {
  color: var(--primary-color);
}

.panel-file-picker svg {
  width: 2.25rem;
  height: 2.25rem;
  border: 2px solid transparent;
  border-radius: 100%;
  transition: all .3s ease-in-out;
}

.file-description {
  color: var(--text-color-light);
}
