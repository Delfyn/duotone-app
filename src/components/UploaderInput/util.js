/* @flow */

import React from 'react';

export function formatFileName(name: string) {
  // eslint-disable-next-line
  return name.length > 10 ? `${name.slice(0, 12)}...` : name;
}

export function formatFileSize(size: number) {
  return (`${Math.round((size / 1024) * 100) / 100} KB`);
}

export function formatFileDescription(name: string, size: number) {
  return (
    <span className="file-description h5">
      {`${formatFileName(name)} ${formatFileSize(size)}`}
    </span>
  );
}
