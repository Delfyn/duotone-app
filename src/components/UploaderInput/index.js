/* @flow */

import React from 'react';
import classNames from 'classnames';
import { formatFileDescription } from './util';
import './style.css';

type Props = {
  openFilepicker: Function,
  fileName?: string,
  fileSize?: number,
};

const UploaderInput = ({ openFilepicker, fileName, fileSize }: Props) => (
  <header
    className={classNames('panel-file-picker', { 'has-file': !!fileName }, 'flex items-center px2 py1')}
    onClick={openFilepicker}
  >
    <svg viewBox="24 0 72 72">
      <path
        d="M60.493 72C79.883 72 96 55.882 96 36.493 96 16.118 79.882 0 60.493 0 40.118 0 24 16.118 24 36.493 24 55.883 40.118 72 60.493 72z"
        fill="#333"
        fillRule="evenodd"
      />
      <path
        d="M58 34h-9c-.553 0-1 .452-1 1.01v1.98c0 .567.448 1.01 1 1.01h9v9c0 .553.452 1 1.01 1h1.98c.567 0 1.01-.448 1.01-1v-9h9c.553 0 1-.452 1-1.01v-1.98c0-.567-.448-1.01-1-1.01h-9v-9c0-.553-.452-1-1.01-1h-1.98c-.567 0-1.01.448-1.01 1v9z"
        fill="#FFF"
        fillRule="evenodd"
      />
    </svg>
    <div className="mx2">
      <h2 className={`${fileName ? 'h4' : 'h3'} regular m0`}>
        {fileName ? 'Change image' : 'Choose image'}
      </h2>
      {fileName && fileSize ? formatFileDescription(fileName, fileSize) : null}
    </div>
  </header>
);

export default UploaderInput;
