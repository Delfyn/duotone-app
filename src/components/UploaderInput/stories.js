/* @flow */

import React from 'react';
import { storiesOf, action } from '@kadira/storybook';
import UploaderInput from './index';

storiesOf('UploaderInput', module)
  .add('with file', () => (
    <UploaderInput
      openFilepicker={action('openFilepicker')}
      fileName="my-image.png"
      fileSize={12345}
    />
  ))
  .add('without file', () => (
    <UploaderInput openFilepicker={action('openFilepicker')} />
  ));
