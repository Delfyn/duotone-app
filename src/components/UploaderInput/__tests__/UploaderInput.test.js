import React from 'react';
import renderer from 'react-test-renderer';
import UploaderInput from '../index';

test('UploaderInput renders correctly', () => {
  const component = renderer.create(<UploaderInput />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('UploaderInput renders file description correctly', () => {
  const component = renderer.create(
    <UploaderInput fileName="my-image.png" fileSize={12345} />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
