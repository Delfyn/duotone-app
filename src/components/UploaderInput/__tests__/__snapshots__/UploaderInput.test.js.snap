exports[`test UploaderInput renders correctly 1`] = `
<header
  className="panel-file-picker flex items-center px2 py1"
  onClick={undefined}>
  <svg
    viewBox="24 0 72 72">
    <path
      d="M60.493 72C79.883 72 96 55.882 96 36.493 96 16.118 79.882 0 60.493 0 40.118 0 24 16.118 24 36.493 24 55.883 40.118 72 60.493 72z"
      fill="#333"
      fillRule="evenodd" />
    <path
      d="M58 34h-9c-.553 0-1 .452-1 1.01v1.98c0 .567.448 1.01 1 1.01h9v9c0 .553.452 1 1.01 1h1.98c.567 0 1.01-.448 1.01-1v-9h9c.553 0 1-.452 1-1.01v-1.98c0-.567-.448-1.01-1-1.01h-9v-9c0-.553-.452-1-1.01-1h-1.98c-.567 0-1.01.448-1.01 1v9z"
      fill="#FFF"
      fillRule="evenodd" />
  </svg>
  <div
    className="mx2">
    <h2
      className="h3 regular m0">
      Choose image
    </h2>
  </div>
</header>
`;

exports[`test UploaderInput renders file description correctly 1`] = `
<header
  className="panel-file-picker has-file flex items-center px2 py1"
  onClick={undefined}>
  <svg
    viewBox="24 0 72 72">
    <path
      d="M60.493 72C79.883 72 96 55.882 96 36.493 96 16.118 79.882 0 60.493 0 40.118 0 24 16.118 24 36.493 24 55.883 40.118 72 60.493 72z"
      fill="#333"
      fillRule="evenodd" />
    <path
      d="M58 34h-9c-.553 0-1 .452-1 1.01v1.98c0 .567.448 1.01 1 1.01h9v9c0 .553.452 1 1.01 1h1.98c.567 0 1.01-.448 1.01-1v-9h9c.553 0 1-.452 1-1.01v-1.98c0-.567-.448-1.01-1-1.01h-9v-9c0-.553-.452-1-1.01-1h-1.98c-.567 0-1.01.448-1.01 1v9z"
      fill="#FFF"
      fillRule="evenodd" />
  </svg>
  <div
    className="mx2">
    <h2
      className="h4 regular m0">
      Change image
    </h2>
    <span
      className="file-description h5">
      my-image.png... 12.06 KB
    </span>
  </div>
</header>
`;
